### PR TITLE
feat!: add support for nuxt 3/nuxt bridge

### PR DESCRIPTION
**Pull request by ricardogobbosouza**
This PR enables this module to work for Nuxt Bridge & Nuxt 3.

## 🔥 Notable changes

- [x] Uses `@nuxt/module-builder` to build module
- [x] Refactor structure using `@nuxt/kit`
- [X] Update docs (#95)
- [x] Re-enable e2e test suite

### CODE REVIEW

File: src/module.ts [hunk 1]
```diff
@@ -32,167 +31,129 @@ const nuxtModule: Module<ModuleOptions> = function (moduleOptions) {
...
-    }
-
-    const googleFontsHelper = new GoogleFontsHelper({
-      families: options.families,
-      display: options.display,
-      subsets: options.subsets,
-      text: options.text
-    })
+      // merge fonts from valid head link
+      const fontsParsed = nuxt.options.head.link!
```
Comment (danielroe) [inline src/module.ts:50]:
```suggestion
      const fontsParsed = nuxt.options.app.head.link!
```

File: src/module.ts [hunk 2]
```diff
@@ -32,167 +31,129 @@ const nuxtModule: Module<ModuleOptions> = function (moduleOptions) {
...
 
-declare module '@nuxt/types' {
-  interface NuxtConfig { [CONFIG_KEY]?: ModuleOptions } // Nuxt 2.14+
-  interface Configuration { [CONFIG_KEY]?: ModuleOptions } // Nuxt 2.9 - 2.13
-}
+      // no-JS fallback
+      nuxt.options.head.noscript!.push({
+        hid: 'gf-noscript',
+        innerHTML: `<link rel="stylesheet" href="${url}">`
+      })
```
Comment (danielroe) [inline src/module.ts:152]: Noscript is not implemented yet for vueuse/head - but see https://github.com/vueuse/head/pull/71.

**Review by danielroe — COMMENTED** [review]
Looking great! Main issue is differences between vue-meta & vueuse/head.

File: src/module.ts [hunk 3]
```diff
@@ -32,167 +31,129 @@ const nuxtModule: Module<ModuleOptions> = function (moduleOptions) {
...
 
-export default nuxtModule
+      // Disable sanitazions
+      nuxt.options.head.__dangerouslyDisableSanitizersByTagID!['gf-script'] = ['innerHTML']
+      nuxt.options.head.__dangerouslyDisableSanitizersByTagID!['gf-noscript'] = ['innerHTML']
```
[the remark this answers is not in the document]
Comment (ricardogobbosouza) [reply src/module.ts:159]: I added support for `vue-meta` and `@vueuse/head`

File: src/module.ts (hunk 2 above)
Comment (ricardogobbosouza) [reply src/module.ts]: Ok, added comment

**Comment by CKGrafico** [comment]
This is in draft but checking your todo list seems ready to be used, can we have an alpha version or something to don't use a github "link" in our package json?

**Review by danielroe — APPROVED** [review]
Looks good to me! I think you could possibly simplify things slightly by wrt nuxt 2/3 and injecting head, maybe with a helper function, if you wanted to. But that's only a nitpick.

**Pull request merged**